tests/unit/specification/parser_test.py

```diff
@@ -383,128 +1012,548 @@ def test_parse_type_derivation_spec() -> None:
                type Bar is new Foo;
             end Test;
         """,
-        {
-            "Test": ast.Specification(
-                ast.ContextSpec([]),
-                ast.PackageSpec(
-                    "Test",
-                    [
-                        model.ModularInteger("__PACKAGE__::T", expr.Number(256)),
-                        ast.MessageSpec("__PACKAGE__::Foo", [ast.Component("N", "T")]),
-                        ast.DerivationSpec("__PACKAGE__::Bar", "Foo"),
+        Model(
+            types=[
+                model.Enumeration(
+                    identifier=ID("__BUILTINS__::Boolean"),
+                    literals=[
+                        (ID("False"), expr.Number(0)),
+                        (ID("True"), expr.Number(1)),
```

Comment:
Previously the resulting AST was tested here, but now the resulting model is tested. It is the only test where a model is used. What is the reason for that?

Reply:
There is no good reason. I changed this back into an AST comparison in f959dc8c.